pulser-core/pulser/noise_model.py

```diff
@@ -366,9 +416,11 @@ def _check_eff_noise(
             if operator.ndim != 2:
                 raise ValueError(f"Operator '{op!r}' is not a 2D array.")
 
-            if operator.shape != (2, 2):
-                raise NotImplementedError(
-                    f"Operator's shape must be (2,2) not {operator.shape}."
+            if operator.shape not in possible_shapes:
+                raise ValueError(
+                    f"With{'' if with_leakage else 'out'} leakage, operator's "
+                    f"shape must be {possible_shapes[0]} or "
+                    f"{possible_shapes[1]}, not {operator.shape}."
```

Comment:
So the changes for the leakage will also allow qutrit operators without leakage, right? I guess this will make a weird in-between state where our NoiseModel supports effective noise on qutrits but our `QutipEmulator` doesn't

Reply:
You are right, I am adding the leakage in effective noise operators in a separate branch from this PR and I think I should modify this test there.